framework/db/oci/Schema.php

```diff
@@ -164,11 +165,42 @@ protected function findColumns($table)
                 $table->primaryKey[] = $c->name;
                 $table->sequenceName = $this->getTableSequenceName($table->name);
             }
+            $c->defaultValue = $this->getColumnDefaultValue($c, $c->defaultValue);
         }
         return true;
     }
 
     /**
+     * Extracts column default value.
+     * @param ColumnSchema $column
+     * @param string $value default value expression
+     * @return mixed null value, a number, string or \yii\db\Expression object
+     */
+    private function getColumnDefaultValue($column, $value)
+    {
+        if (!strcasecmp($value, 'null') || $value === '' || $value === null) {
+            return null;
+        }
+        $firstChar = strtolower($value{0});
+        $secondChar = strlen($value) > 1 ? strtolower($value{1}) : null;
+        $isQuoted = $firstChar === "'" || ($firstChar === 'n' && $secondChar === "'");
+        $isNumeric = $firstChar === '.' || ('0' <= $firstChar && $firstChar <= '9')
+            || (($firstChar === '-' || $firstChar === '+') && '0' <= $secondChar && $secondChar <= '9');
+        //! @todo DATE typecasts should be detected here
```

Comment:
Yes, I'll mark the PR as WIP. All dates must be string literals with type casts. It would be very useful to extract them as strings in the applications but the type casts must be stripped. It will add a considerable amout of code here. I need some opinions on that first before writing it.